src/renderer/services/eth2/client/lighthouse/validator.ts

```diff
@@ -0,0 +1,123 @@
+import {IValidatorApi} from "@chainsafe/lodestar-validator/lib/api/interface/validators";
+import {HttpClient} from "../../../api";
+import {IBeaconConfig} from "@chainsafe/eth2.0-config";
+import {IBeaconClientOptions} from "../interface";
+import {Attestation, BeaconBlock, BLSPubkey, SignedBeaconBlock, Slot, ValidatorDuty} from "@chainsafe/eth2.0-types";
+import {ILighthouseDutiesRequest, ILighthouseDutiesResponse} from "./types";
+import {toHexString} from "../../../utils/crypto";
+import {fromHex} from "../../../utils/bytes";
+import {computeEpochAtSlot} from "@chainsafe/eth2.0-state-transition";
+import {fromJson, toJson} from "@chainsafe/eth2.0-utils";
+
+export enum LighthouseValidatorRoutes {
+
+    DUTIES = "/validator/duties",
+    ATTESTATION = "/validator/attestation",
+    BLOCK = "/validator/block"
+
+}
+
+export class LighthouseValidatorApiClient implements IValidatorApi {
+
+    private client: HttpClient;
+    private config: IBeaconConfig;
+    private trackedValidators: Set<BLSPubkey> = new Set();
+
+    public constructor(options: IBeaconClientOptions, validators: BLSPubkey[] = []) {
+        this.client = new HttpClient(options.urlPrefix);
+        this.config = options.config;
+        validators.forEach((validator) => this.trackedValidators.add(validator));
+    }
+    
+    public async getAttesterDuties(epoch: number, validatorPubKeys: BLSPubkey[]): Promise<ValidatorDuty[]> {
+        validatorPubKeys.forEach((key) => this.trackedValidators.add(key));
+        const response = await this.client.post<ILighthouseDutiesRequest, ILighthouseDutiesResponse[]>(
+            LighthouseValidatorRoutes.DUTIES,
+            {
+                epoch,
+                pubkeys: validatorPubKeys.map(toHexString)
+            });
+        return validatorPubKeys.map((validatorPubKey) => {
+            const lhDuty = response.find((value => fromHex(value.validator_pubkey).equals(validatorPubKey)));
+            if(lhDuty) {
+                return {
+                    validatorPubkey: validatorPubKey,
+                    attestationSlot: lhDuty.attestation_slot,
+                    committeeIndex: lhDuty.attestation_committee_index
+                } as ValidatorDuty;
+            } else {
+                return null;
+            }
+        }).filter((value) => !!value) as ValidatorDuty[];
```

Comment:
Are you filtering out `null` values? It's hacky and confusing, rather not return null (not use map in that case)

Reply:
What would you use?
This is somewhat common practicebI guess

Reply:
class for loop? Idk, I am not sure why would iterating something twice be a common practice xD Plus, looks weird that you return something and then filter it.

Reply:
It will probably get inlined by node engine and I don't think it's gonna matter as it will mostly be 1 or 2 elem array :smile:

Reply:
I was pointing to more readable and junior-friendly code rather than cooler code :P